tests/unit/mathematics/linalg/SGLinalg_unittest.cc

```diff
@@ -0,0 +1,72 @@
+#include <shogun/lib/config.h>
+
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+
+#ifdef HAVE_VIENNACL
+#include <shogun/mathematics/linalg/LinalgBackendViennaCL.h>
+#endif
+
+#include <shogun/lib/SGVector.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+
+TEST(LinalgRefactor, linalg_cpu_backend_dot)
+{
+	const index_t size = 10;
+	SGVector<int32_t> a(size), b(size);
+	a.set_const(1);
```

Comment:
I  meant: please dont use vectors like [1,1,1,1,1] in unit test. Choose something like [1,2,3,4,5] or random